app/presenters/result_set_presenter.rb

```diff
@@ -28,23 +28,15 @@ def search_results_content
       zero_results: total.zero?,
       page_count: documents.count,
       finder_name: finder.name,
-      debug_score: debug_score,
+      debug_score: debug_score
     }
   end
 
   def documents
     @documents ||= begin
       results.each_with_index.map do |result, index|
         metadata = metadata_presenter_class.new(result.metadata).present
-        doc = SearchResultPresenter.new(result, metadata).to_hash
-        if  index === 0 && highlight_top_result?
-          doc[:top_result] = true
-          doc[:summary] = result.truncated_description
-        end
-        {
-          document: doc,
-          document_index: index + 1
-        }
+        SearchResultPresenter.new(result, metadata, index, results.count, finder.name, debug_score, highlight(index)).to_hash
```

Comment:
can we move index and finder.name into document? - document already has finder as a member so should be straightforward to delegate

Reply:
I agree that would be a good idea but it's used in a number of other places and I don't want to start introducing more changes in an already complex PR. Can we look at it in a separate PR?